auth/credentials/detect.go

```diff
@@ -250,3 +253,31 @@ func clientCredConfigFromJSON(b []byte, opts *DetectOptions) *auth.Options3LO {
 		AuthStyle: auth.StyleInParams,
 	}
 }
+
+// GetGoogleMtlsTokenURL returns the mTLS token exchange URL
+func GetGoogleMtlsTokenURL() string {
+	return googleMtlsTokenURL
+}
+
+// CustomHTTPClient constructs an HTTPClient using the provided tlsConfig, to support mTLS.
+func CustomHTTPClient(tlsConfig *tls.Config) *http.Client {
```

Comment:
Why is this needed if a user can provide a http.Client already.

Also, this feels like if we were going to add it it should be done in the httptransport package.

Reply:
looks like it is use in two package. I think this should be in internal.

Reply:
Moved to internal transport.go